datadriven/src/datadriven.rs

```diff
@@ -0,0 +1,217 @@
+use crate::test_data::TestData;
+use crate::test_data_reader::TestDataReader;
+use anyhow::Result;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+/// The main function to run tests
+///
+/// You need to pass the path of `testdata` where store the test cases, and your function
+/// to output expected result.
+///
+/// It will be compare with crate `difference`
+///
+/// It invokes a data-driven test. The test cases are contained in a
+/// separate test file and are dynamically loaded, parsed, and executed by this
+/// testing framework. By convention, test files are typically located in a
+/// sub-directory called "testdata". Each test file has the following format:
+///
+/// ```txt
+/// <command> [arg | arg=val | arg=(val1, val2, ...)]...
+/// <input to the command>
+/// ----
+/// <expected results>
+/// ````
+///
+/// The command input can contain blank lines. However, by default, the expected
+/// results cannot contain blank lines. This alternate syntax allows the use of
+/// blank lines:
+/// ```txt
+/// <command> [arg | arg=val | arg=(val1, val2, ...)]...
+/// <input to the command>
+/// ----
+/// ----
+/// <expected results>
+///
+/// <more expected results>
+/// ----
+/// ----
+/// ````
+/// Note: blank line is required after the second *`----`*.
```

Comment:
Do you mean that when using two `---`, there has to be at least one blank line in expected result?

Reply:
yes, and it is also true when using one separator.

Reply:
I don't get it. The docs say when using one separator, blank line is not allow. Can you give an example about the illegal case described by " blank line is required after the second *`----`*."?